### PR TITLE
Added "External Project ID" column to the Projects page

### PR DESCRIPTION
**Pull request by samouzegar10**

### CODE REVIEW

**Comment by UlysseFG** [comment]
The search for the column is not working ...

File: frontend/src/components/projects/ProjectsTableColumns.tsx [hunk 1]
```diff
@@ -128,6 +141,7 @@ export const PROJECT_FILTERS: { [key in ProjectColumnID]: FilterDescription } =
 export const PROJECT_FILTER_KEYS: { [key in ProjectColumnID]: string } = {
 	[ProjectColumnID.ID]: 'id',
 	[ProjectColumnID.NAME]: 'name',
+    [ProjectColumnID.EXTERNAL_ID]: 'external_id',
```
Comment (UlysseFG) [inline frontend/src/components/projects/ProjectsTableColumns.tsx:144]: try to indent the same as the stuff around ;)

File: frontend/src/components/projects/ProjectsTableColumns.tsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { UNDEFINED_FILTER_KEY } from '../pagedItemsTable/PagedItemsFilters'
 export enum ProjectColumnID {
 	ID = 'ID',
 	NAME = 'NAME',
+    EXTERNAL_ID = 'EXTERNAL_ID',
```
Comment (UlysseFG) [inline frontend/src/components/projects/ProjectsTableColumns.tsx:12]: indent ...

File: frontend/src/components/projects/ProjectsListContent.tsx [hunk 3]
```diff
@@ -23,6 +23,7 @@ import { sort } from 'rambda'
 const projectsListContentColumns = [
 	PROJECT_COLUMN_DEFINITIONS.ID,
 	PROJECT_COLUMN_DEFINITIONS.NAME,
+    PROJECT_COLUMN_DEFINITIONS.EXTERNAL_ID,
```
Comment (UlysseFG) [inline frontend/src/components/projects/ProjectsListContent.tsx:26]: indent ...

**Comment by UlysseFG** [comment]
Since we now display the external Id in the frontend, it might be good to add the external id to the project export serializer too to provide this information in projects excel dumps ...

**Comment by samouzegar10** [comment]
> Since we now display the external Id in the frontend, it might be good to add the external id to the project export serializer too to provide this information in projects excel dumps ...

it can be added to the redmine and pushed another pr

**Review by nafiz1001 — APPROVED** [review]

**Pull request merged**